src/structure.js

```diff
@@ -68,6 +77,57 @@ Structure.prototype.cursor = function (path) {
   return Cursor.from(self.current, path, changeListener);
 };
 
+Structure.prototype.reference = function (path) {
+  var self = this, pathId = pathString(path);
+  var listenerNs = self._pathListeners[pathId];
+
+  var cursor = this.cursor(path);
+  var changeListener = function (newRoot, oldRoot, changedPath) { cursor = self.cursor(path); };
+
+  var referenceListeners = [changeListener];
+  this._pathListeners[path] = !listenerNs ? [changeListener] : listenerNs.concat(changeListener);
+
+  return {
+    onChange: function (newFn) {
+      if (typeof newFn !== 'function') return;
+      var fnIndex, listenerNs = self._pathListeners[pathId];
+
+      self._pathListeners[pathId] = !listenerNs ? [newFn] : listenerNs.concat(newFn);
+      fnIndex = self._pathListeners[pathId].indexOf(newFn);
+
+      referenceListeners = (referenceListeners || []).concat(newFn);
+      localListenerIndex = referenceListeners.indexOf(newFn);
+
+      return function () {
+        if (referenceListeners[localListenerIndex] &&
+          referenceListeners[localListenerIndex] === newFn) {
+          referenceListeners.splice(localListenerIndex, 1);
+        }
+
+        if (!self._pathListeners[pathId]) return;
+        if (!self._pathListeners[pathId][fnIndex] ||
+          self._pathListeners[pathId][fnIndex] !== newFn) return;
+        self._pathListeners[pathId].splice(fnIndex, 1);
+      };
+    },
+    cursor: function () {
+      return cursor;
+    },
+    unsubscribeAll: function () {
+      cursor = void 0;
```

Comment:
IMO, this is unnecessary, `changeListener` (assuming it's never removed in any array), would keep `cursor` fresh.